test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_binary_size.phpt

```diff
@@ -0,0 +1,279 @@
+--TEST--
+Test for retrieving encrypted data of binary types with different sizes as output parameters
+--DESCRIPTION--
+Test implicit conversions between different binary types of different sizes
+With Always Encrypted, implicit conversion works if:
+1. From a binary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m == n
+2. From a binary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m == n
+3. From a varbinary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m == n
+4. From a varbinary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m == n
+Without AlwaysEncrypted, implicit conversion works if:
+1. From a binary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m, n == any value
+2. From a binary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m <= n (exclude SQLSRV_SQLTYPE_VARBINARY('max'))
+3. From a varbinary(m) column to a SQLSRV_SQLTYPE_BINARY(n) output parameter where m, n == any value
+4. From a varbinary(m) column to a SQLSRV_SQLTYPE_VARBINARY(n) output parameter where m, n == any value
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$dataTypes = array("binary", "varbinary", "varbinary(max)");
+$lengths = array(1, 8, 64, 512, 4000);
+$sqlTypes = array("SQLSRV_SQLTYPE_BINARY", "SQLSRV_SQLTYPE_VARBINARY", "SQLSRV_SQLTYPE_VARBINARY('max')");
+$sqltypeLengths = $lengths;
```

Comment:
why do we need another variable?

Reply:
`$length` cannot be iterated twice in a nested foreach loop